### ISSUE TITLE
Add useful toString to Violation

### ISSUE THREAD

**Issue by Diagoras** [wix/accord#78]
It would be nice to have a predefined and straightforward `toString` method on the `Violation` subtypes so that it'd be easy to get a String representation of the violation for debugging or creating a 400 response. For example:

```scala
RuleViolation(value: null, constraint: "is null", description: Explicit(description: "id")).toString == "id is null"
```

**Comment by holograph** [comment]
What worries me about this is that it might indicate to users that violations are intended to be rendered this way (if there's an API, someone will inevitably use it). In what way isn't the usual case class breakdown sufficient for debugging?

**Comment by Diagoras** [comment]
You're right that the normal case class `toString` works for debugging, but I was more thinking of the kind of verbose, human readable output that you'd use for logging, HTTP responses, etc. If you're concerned about overriding the default `toString`, we could always change it to `toVerboseString` or something. That work?

**Comment by Diagoras** [comment]
I like it! That tree view is great, thanks.
